File: app_bmodule/src/main/java/com/pandaq/app_bmodule/app/lifecycle/AppLifeCycle.java
package com.pandaq.app_bmodule.app.lifecycle;

import android.app.Application;
import android.content.Context;

import com.pandaq.appcore.framework.app.lifecycle.IAppLifeCycle;
import com.pandaq.appcore.utils.log.PLogger;

import androidx.annotation.NonNull;

/**
 * Created by huxinyu on 2018/12/25.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :this module' appLifeCycle impl.the methods will called in global application
 */
public class AppLifeCycle implements IAppLifeCycle {

    @Override
    public void attachBaseContext(@NonNull Context base) {

    }

    @Override
    public void onCreate(@NonNull Application application) {
        PLogger.d("AppInit", "----BModule");
    }

    @Override
    public void onTerminate(@NonNull Application application) {

    }
}

File: app/src/main/java/com/pandaq/pandamvp/base/app/AppLifecycle.kt
package com.pandaq.pandamvp.base.app

import android.app.Application
import android.content.Context
import com.alibaba.android.arouter.launcher.ARouter
import com.pandaq.appcore.BuildConfig
import com.pandaq.appcore.framework.app.lifecycle.IAppLifeCycle

/**
 * Created by huxinyu on 2019/7/8.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :
 */
class AppLifecycle : IAppLifeCycle {
    override fun attachBaseContext(base: Context) {

    }

    override fun onCreate(application: Application) {

    }

    override fun onTerminate(application: Application) {

    }
}